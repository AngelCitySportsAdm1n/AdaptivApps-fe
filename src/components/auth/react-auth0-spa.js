import React, { useState, useEffect, useContext } from 'react';
import createAuth0Client from '@auth0/auth0-spa-js';
import PropTypes from 'prop-types';

const DEFAULT_REDIRECT_CALLBACK = () =>
  window.history.replaceState({}, document.title, window.location.pathname);

export const Auth0Context = React.createContext();
export const useAuth0 = () => useContext(Auth0Context);
//console.log('console.log useAuth0', useAuth0.getTokenSilently);
export const Auth0Provider = ({
  children,
  onRedirectCallback = DEFAULT_REDIRECT_CALLBACK,
  ...initOptions
}) => {
  const [isAuthenticated, setIsAuthenticated] = useState();
  const [user, setUser] = useState();
  const [auth0Client, setAuth0] = useState();
  const [loading, setLoading] = useState(true);
  const [popupOpen, setPopupOpen] = useState(false);

  useEffect(() => {
    const initAuth0 = async () => {
      const auth0FromHook = await createAuth0Client(initOptions);
      setAuth0(auth0FromHook);

      if (
        window.location.search.includes('code=') &&
        window.location.search.includes('state=')
      ) {
        const { appState } = await auth0FromHook.handleRedirectCallback();
        onRedirectCallback(appState);
      }
      const tokenP = auth0FromHook.getIdTokenClaims();

      const isAuthenticated = await auth0FromHook.isAuthenticated();
      //console.log(
      //'Showing this auth0FrameHook',
      //   auth0FromHook.getIdTokenClaims()
      // );
      // checks to make sure user is authenticated and sets user if true.
      setIsAuthenticated(isAuthenticated);

      if (isAuthenticated) {
        const token = await tokenP;
        //console.log('Showing tokenP', token.__raw);
        //localStorage.setItem('golden', token.__raw);
        const user = await auth0FromHook.getUser();
        setUser(user);
        //console.log('authenticated', user);
      }

      setLoading(false);
    };
    initAuth0();
    // eslint-disable-next-line
  }, []);

  const loginWithPopup = async (params = {}) => {
    setPopupOpen(true);
    try {
      await auth0Client.loginWithPopup(params);
    } catch (error) {
      //console.error(error);
    } finally {
      setPopupOpen(false);
    }
    const user = await auth0Client.getUser();
    setUser(user);
    setIsAuthenticated(true);
    //console.log('user', user);
  };

  const handleRedirectCallback = async () => {
    setLoading(true);
    await auth0Client.handleRedirectCallback();
    const user = await auth0Client.getUser();
    setLoading(false);
    setIsAuthenticated(true);
    setUser(user);
    //console.log('redirect', user);
  };
  return (
    <Auth0Context.Provider
      value={{
        isAuthenticated,
        user,
        loading,
        popupOpen,
        loginWithPopup,
        handleRedirectCallback,
        getIdTokenClaims: (...p) => auth0Client.getIdTokenClaims(...p),
        loginWithRedirect: (...p) => auth0Client.loginWithRedirect(...p),
        getTokenSilently: (...p) => auth0Client.getTokenSilently(...p),
        getTokenWithPopup: (...p) => auth0Client.getTokenWithPopup(...p),
        logout: (...p) => auth0Client.logout(...p),
      }}
    >
      {children}
    </Auth0Context.Provider>
  );
};

Auth0Provider.propTypes = {
  children: PropTypes.any,
  onRedirectCallback: PropTypes.any,
};
