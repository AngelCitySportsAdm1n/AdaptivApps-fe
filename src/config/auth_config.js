module.exports = {
  domain: process.env.REACT_APP_CLIENT_DOMAIN,
  clientId: process.env.REACT_APP_CLIENT_ID,
  roleUrl: 'http://adaptivapps.com/roles',
  audience: process.env.REACT_APP_AUDIENCE,
  responseType: 'token id_token',
  scope: 'openid email',
};
