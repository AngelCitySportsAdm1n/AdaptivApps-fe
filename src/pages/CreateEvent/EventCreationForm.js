import React, { useState } from 'react';
import { Link } from '@reach/router';
import { useForm } from 'react-hook-form';
import { useMutation } from 'react-apollo';
import { CREATE_EVENT } from './queries/EventsQuery';
import {
  Box,
  Form,
  Text,
  Flex,
  Input,
  useModal,
  Modal,
  Button,
  theme,
} from 'adaptiv-ui';

// This is the form being used in to create an event
const EventCreationForm = () => {
  const [isActive, toggle] = useModal();

  const [currEvent, setCurrEvent] = useState({
    id: '',
    title: '',
    startDate: '',
    endDate: '',
    location: '',
  });

  const [CreateEvent] = useMutation(CREATE_EVENT);

  const { handleSubmit, register } = useForm();

  // creates an event
  const onSubmit = async (values, e) => {
    e.preventDefault();
    const { data } = await CreateEvent({
      variables: {
        title: values.title,
        startDate: values.startDate,
        endDate: values.endDate,
        location: values.location,
        imgUrl: values.imgUrl,
        details: values.details,
      },
    });
    await setCurrEvent(data.createEvent);
    toggle();
  };

  return (
    <Flex ai_start col stretch m="0 0 0 2rem">
      <Text xlf bold mm>
        Create an Event
      </Text>
      <Box h="0.2rem" w="90%" bg="lightgrey" />
      <Box h="2rem" />
      <Form ai_start col stretch onSubmit={handleSubmit(onSubmit)}>
        <Text mf>Event Title</Text>
        <Flex ai_center>
          <Input
            type="text"
            w="25rem"
            name="title"
            ref={register({
              required: 'Required',
            })}
          />
        </Flex>

        <Text mf>Start Date</Text>
        <Flex ai_center>
          <Input
            type="date"
            w="25rem"
            name="startDate"
            ref={register({
              required: 'Required',
            })}
          />
        </Flex>

        <Text mf>End Date</Text>
        <Flex ai_center>
          <Input
            type="date"
            w="25rem"
            name="endDate"
            ref={register({
              required: 'Required',
            })}
          />
        </Flex>

        <Text mf>Location</Text>
        <Flex ai_center>
          <Input
            type="text"
            w="25rem"
            name="location"
            ref={register({
              required: 'Required',
            })}
          />
        </Flex>

        <Text mf>Image Url</Text>
        <Flex ai_center>
          <Input type="text" w="25rem" name="imgUrl" ref={register()} />
        </Flex>

        <Text mf>Details</Text>
        <Flex ai_center>
          <Input type="text" w="25rem" name="details" ref={register()} />
        </Flex>

        <Button
          jc_center
          secondary
          border={`2px solid ${theme.primary}`}
          w="9rem"
          h="4rem"
          type="submit"
        >
          Submit
        </Button>
      </Form>
      <Modal isActive={isActive}>
        <Link to={`/events/create/${currEvent.id}`}>
          Event Created. Add Activities.
        </Link>
      </Modal>
    </Flex>
  );
};

export default EventCreationForm;
