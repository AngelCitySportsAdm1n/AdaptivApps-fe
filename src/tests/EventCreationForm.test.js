import React from "react";
import EventCreationForm from "../pages/CreateEvent/EventCreationForm";
import renderer from "react-test-renderer";

import { ApolloProvider } from "@apollo/react-hooks";
import ApolloClient from "apollo-boost";
import { getIdTokenClaims } from "../config/react-auth0-spa";

const client = new ApolloClient({
  uri: process.env.REACT_APP_API_URL,
  credentials: "same-origin",
  request: async operation => {
    const token = await getIdTokenClaims();
    // Attach token to headers
    operation.setContext(context => ({
      headers: {
        ...context.headers,
        Authorization: token.__raw,
      },
    }));
  },
});

it("renders event creation form correctly", () => {
  const tree = renderer
    .create(
      <ApolloProvider client={client}>
        <EventCreationForm />
      </ApolloProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
