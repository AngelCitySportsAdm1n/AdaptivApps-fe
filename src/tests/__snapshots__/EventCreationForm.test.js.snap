// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders event creation form correctly 1`] = `
<div
  className="sc-ifAKCX fpFZyH"
>
  <h1>
    Create an Event
  </h1>
  <div
    className="sc-iwsKbI jOijdN"
  />
  <div
    className="sc-iwsKbI bqkBdz"
  />
  <form
    className="sc-EHOje gLPZAQ"
    onSubmit={[Function]}
  >
    <div
      className="sc-gqjmRU hCVKYo"
    >
      Event Title
    </div>
    <div
      className="sc-ifAKCX jLfGdD"
    >
      <input
        className="sc-bZQynM dDlBhs"
        name="title"
        type="text"
      />
    </div>
    <div
      className="sc-gqjmRU hCVKYo"
    >
      Start Date
    </div>
    <div
      className="sc-ifAKCX jLfGdD"
    >
      <input
        className="sc-bZQynM dDlBhs"
        name="startDate"
        type="date"
      />
    </div>
    <div
      className="sc-gqjmRU hCVKYo"
    >
      End Date
    </div>
    <div
      className="sc-ifAKCX jLfGdD"
    >
      <input
        className="sc-bZQynM dDlBhs"
        name="endDate"
        type="date"
      />
    </div>
    <div
      className="sc-gqjmRU hCVKYo"
    >
      Location
    </div>
    <div
      className="sc-ifAKCX jLfGdD"
    >
      <input
        className="sc-bZQynM dDlBhs"
        name="location"
        type="text"
      />
    </div>
    <div
      className="sc-gqjmRU hCVKYo"
    >
      Image Url
    </div>
    <div
      className="sc-ifAKCX jLfGdD"
    >
      <input
        className="sc-bZQynM dDlBhs"
        name="imgUrl"
        type="text"
      />
    </div>
    <div
      className="sc-gqjmRU hCVKYo"
    >
      Details
    </div>
    <div
      className="sc-ifAKCX jLfGdD"
    >
      <input
        className="sc-bZQynM dDlBhs"
        name="details"
        type="text"
      />
    </div>
    <button
      className="sc-gzVnrw klnqKh"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
